LabExT/View/Movement/CoordinatePairingsWindow.py

```diff
@@ -182,29 +143,76 @@ def _select_device_frame(self):
             )
             self._clear_device_button.pack(side=LEFT, padx=5)
 
-    #
-    #   Callbacks
-    #
+        frame = CustomFrame(self._main_frame)
+        frame.pack(side=TOP, fill=X, pady=5)
+
+        step_hint = Label(frame, text=self._get_coupling_hint())
+        step_hint.pack(side=TOP, fill=X)
+
+        if self.with_input_stage:
+            self._calibration_pairing_widget(
+                frame, self.mover.input_calibration).pack(
+                side=TOP, fill=X)
+
+        if self.with_output_stage:
+            self._calibration_pairing_widget(
+                frame, self.mover.output_calibration).pack(
+                side=TOP, fill=X)
+
+    def __reload__(self) -> None:
+        """
+        Reloads window.
+        """
+        for child in self._main_frame.winfo_children():
+            child.forget()
+
+        self._finish_button.config(state=NORMAL if self._device else DISABLED)
+
+        self.__setup__()
+        self.update_idletasks()
+
+    def finish(self) -> None:
+        """
+        Callback, when user wants to finish the pairing.
+
+        Builds the pairings and calls the on_finish callback.
+
+        Destroys the window.
+        """
+        if self._device is None:
+            messagebox.showwarning(
+                'Device Needed',
+                'Please select a device to create a pairing.',
+                parent=self)
+            return
+
+        pairings = []
+
+        if self.with_input_stage:
+            pairings.append(self._get_pairing(self.mover.input_calibration))
+
+        if self.with_output_stage:
+            pairings.append(self._get_pairing(self.mover.output_calibration))
+
+        self.on_finish(pairings)
+        self.destroy()
+
+    def cancel(self) -> None:
+        """
+        Callback, when user wants to cancel the pairing.
+
+        Resets the device.
 
-    def _finish(self):
-        try:
-            if self._in_calibration:
-                self._in_stage_coordinate = self._in_calibration.stage.get_current_position()
-            if self._out_calibration:
-                self._out_stage_coordinate = self._out_calibration.stage.get_current_position()
-
-            self.destroy()
-        except StageError as e:
-            messagebox.showerror(
-                "Error", "Could not get current position: {}".format(e))
-
-    def _cancel(self):
-        self._in_stage_coordinate = None
-        self._out_stage_coordinate = None
+        Destroys the window.
+        """
         self._device = None
         self.destroy()
 
-    def _on_device_selection(self):
+    #
+    #   Callbacks
+    #
+
+    def _on_device_selection(self) -> None:
```

Comment:
I am missing the "move approximately to device" function here. What happened to it?

Reply:
Yep that's right, will add the function back when the mover is implemented the `move_to_device` function.

Reply:
Okay, so will this be part of this PR (so I wait with merging) or a future PR?

Reply:
I will do in a future PR (after we have collision avoidance merged)